packages/sdk/src/actor/actor.ts

```diff
@@ -632,83 +631,28 @@ export class Actor implements ActorLike, Patchable<ActorLike> {
-	 * When an animation is disabled, it is also paused (its time does not move forward).
-	 * @param animationName The name of the animation.
-	 */
-	public disableAnimation(animationName: string) {
-		this.setAnimationState(animationName, { enabled: false });
-	}
-
-	/**
-	 * @deprecated Set [[Animation.isPlaying]] instead.
-	 * Starts the animation (sets animation speed to 1).
-	 * @param animationName The name of the animation.
-	 */
-	public resumeAnimation(animationName: string) {
-		this.setAnimationState(animationName, { enabled: true });
-	}
-
-	/**
-	 * @deprecated Set [[Animation.isPlaying]] instead.
-	 * Stops the animation (sets animation speed to zero).
-	 * @param animationName The name of the animation.
-	 */
-	public pauseAnimation(animationName: string) {
-		this.setAnimationState(animationName, { enabled: false });
-	}
-
-	/**
-	 * @deprecated Set [[Animation.time]] instead.
-	 * Sets the animation time (units are in seconds).
-	 * @param animationName The name of the animation.
-	 * @param time The desired animation time. A negative value seeks to the end of the animation.
-	 */
-	public setAnimationTime(animationName: string, time: number) {
-		this.setAnimationState(animationName, { time });
-	}
-
-	/**
-	 * @deprecated Set properties of an [[Animation]] instance instead.
-	 * (Advanced) Sets the time, speed, and enabled state of an animation.
-	 * @param animationName The name of the animation.
-	 * @param options The time, speed and enabled state to apply. All values are optional. Only the values
-	 * provided will be applied.
+	 * @deprecated
+	 * Use [[AssetContainer.createAnimationData]] and [[AnimationData.bind]] instead.
 	 */
-	public setAnimationState(animationName: string, state: SetAnimationStateOptions) {
-		return this.context.internal.setAnimationState(this.id, animationName, state);
+	public createAnimation(animationName: string, options: any) {
+		throw new Error('Actor.createAnimation has been removed. Use AssetContainer.createAnimationData ' +
+			'and AnimationData.bind instead.');
 	}
 
 	/**
-	 * Animate actor properties to the given value, following the specified animation curve. Actor transform
-	 * is the only animatable property at the moment. Other properties such as light color may become animatable
-	 * in the future.
-	 * @param value The desired final state of the animation.
+	 * @deprecated
+	 * Use [[Animation.AnimateTo]] instead.
+	 * @param value The desired final state of the actor.
 	 * @param duration The length of the interpolation (in seconds).
 	 * @param curve The cubic-bezier curve parameters. @see AnimationEaseCurves for predefined values.
 	 */
 	public animateTo(value: Partial<ActorLike>, duration: number, curve: number[]) {
-		this.context.internal.animateTo(this.id, value, duration, curve);
+		// added this in because it's easy. not so for createAnimation above.
+		Animation.AnimateTo(this.context, this as Actor, {
```

Comment:
Return the promise here?

Reply:
The old method returned `void`, so we should keep that prototype, right?

Reply:
I think returning the promise would be an improvement, would you agree? It would allow the caller to await on the completion of the animation.